src/integrationTest/java/org/opentripplanner/base/ResponseStructureIT.java

```diff
@@ -0,0 +1,170 @@
+
+        for (int i = 0; i < itinerary.legs.size() - 1; ++i) {
+            assertThat(itinerary.legs.get(i).to, equalTo(itinerary.legs.get(i + 1).from));
+        }
+    }
+
+    private void assertItinerary1(Itinerary itinerary) {
+        assertThat(itinerary.itineraryType, equalTo("WALK+MOTORBIKE"));
+        assertThat(itinerary.usedNotRecommendedRoute, equalTo(false)); // the best route?
+
+        assertThat(itinerary.distanceTraversedInMode, Matchers.hasEntry(TraverseMode.CAR, 3778.4299999999994)); // MOTORBIKE is a CAR
+        assertThat(itinerary.distanceTraversedInMode, Matchers.hasEntry(TraverseMode.WALK, 585.158));
+        assertThat(itinerary.traverseDistance, equalTo(3778.4299999999994 + 585.158));
+
+        assertThat(itinerary.timeTraversedInMode, Matchers.hasEntry(TraverseMode.CAR, 942));
+        assertThat(itinerary.timeTraversedInMode, Matchers.hasEntry(TraverseMode.WALK, 470));
+        assertThat(itinerary.duration, equalTo((long) 942 + 470));
+
+        assertThat(itinerary.legs.size(), equalTo(2));
+
+        assertThat(itinerary.legs.get(0).mode, equalTo(TraverseMode.WALK));
+        assertThat(itinerary.legs.get(0).agencyName, equalTo(null));
+
+        assertThat(itinerary.legs.get(1).mode, equalTo(TraverseMode.CAR));
+        assertThat(itinerary.legs.get(1).vehicleDescription, notNullValue());
+        assertThat(itinerary.legs.get(1).vehicleDescription.getVehicleType(), equalTo(VehicleType.MOTORBIKE));
+        assertThat(itinerary.legs.get(1).vehicleDescription.getProvider().getProviderName(), equalTo("Blinkee"));
+        assertThat(itinerary.legs.get(1).vehicleDescription.getLatitude(), closeTo(itinerary.legs.get(1).from.lat, 0.001)); // vehicle position is not exactly on route
+        assertThat(itinerary.legs.get(1).vehicleDescription.getLongitude(), closeTo(itinerary.legs.get(1).from.lon, 0.001));
+
+        for (int i = 0; i < itinerary.legs.size() - 1; ++i) {
+            assertThat(itinerary.legs.get(i).to, equalTo(itinerary.legs.get(i + 1).from));
+        }
+    }
+
+    private void assertItinerary0(Itinerary itinerary) {
+        assertThat(itinerary.itineraryType, equalTo("WALK+KICKSCOOTER+TRANSIT"));
+        assertThat(itinerary.usedNotRecommendedRoute, equalTo(true));
+
+        assertThat(itinerary.distanceTraversedInMode, Matchers.hasEntry(TraverseMode.BICYCLE, 2646.726000000001)); // MOTORBIKE is a CAR
+        assertThat(itinerary.distanceTraversedInMode, Matchers.hasEntry(TraverseMode.WALK, 134.779));
+        assertThat(itinerary.traverseDistance, equalTo(2646.726000000001 + 134.779));
+
+        assertThat(itinerary.timeTraversedInMode, Matchers.hasEntry(TraverseMode.BICYCLE, 832));
+        assertThat(itinerary.timeTraversedInMode, Matchers.hasEntry(TraverseMode.WALK, 121));
+        assertThat(itinerary.timeTraversedInMode, Matchers.hasEntry(TraverseMode.TRANSIT, 240));
+        assertThat(itinerary.duration, equalTo((long) (832 + 121 + 240)));
```

Comment:
Czy na pewno chcemy trzmać zahardkodowane czasy przejścia nogami?

Reply:
idea jest taka, że wszystko jest deterministyczne, więc jak coś się zmieni to będziemy o tym wiedzieć. Trochę pracy będzie z poprawianiem tesów za każdym razem, ale na ten moment jest tylko jeden więc bez przesady.